CharlieBackend.Api.UnitTes/LessonServiceTests.cs

```diff
@@ -11,20 +11,87 @@
+            _themeRepositoryMock = new Mock<IThemeRepository>();
+            _mentorRepositoryMock = new Mock<IMentorRepository>();
+            _studentGroupRepositoryMock = new Mock<IStudentGroupRepository>();
+            _visitRepositoryMock = new Mock<IVisitRepository>();
+        }
+        private static StudentGroup CreateLessonStudentGroup(out List<VisitDto> visitDto, out LessonDto createdLesson, out CreateLessonDto createLessonDto)
+        {
+            List<StudentOfStudentGroup> studentOfStudentGroup = new List<StudentOfStudentGroup>
+            {
+                new StudentOfStudentGroup { StudentGroupId =3 , StudentId = 11 },
+                new StudentOfStudentGroup { StudentGroupId =3 , StudentId = 14 }
+            };
+            StudentGroup studentGroup = new StudentGroup() { Id = 3, StudentsOfStudentGroups = studentOfStudentGroup };
+            DateTime lessonDate = DateTime.Parse("2020-11-18T15:00:00.384Z");
+            visitDto = new List<VisitDto>
+            {
+                new VisitDto()
+                {
+                    StudentId = 11,
+                    Presence = true
+                },
+
+                new VisitDto()
+                {
+                    StudentId = 14,
+                    Presence = false
+                }
+            };
+            createdLesson = new LessonDto()
+            {
+                Id = 7,
+                ThemeName = "ExampleName",
+                MentorId = 2,
+                StudentGroupId = 3,
+                LessonDate = lessonDate,
+                LessonVisits = visitDto
+            };
+            createLessonDto = new CreateLessonDto
+            {
+                ThemeName = "ExampleName",
+                MentorId = 2,
+                StudentGroupId = 3,
+                LessonDate = lessonDate,
+                LessonVisits = visitDto
+            };
+            return studentGroup;
+        }
+        private void MockEntities(StudentGroup studentGroup)
+        {
+            _studentGroupRepositoryMock.Setup(x => x.GetByIdAsync(studentGroup.Id)).ReturnsAsync(studentGroup);
+            _studentGroupRepositoryMock.Setup(x => x.GetGroupStudentsIds(studentGroup.Id)).ReturnsAsync(new List<long?> { 11, 14 });
+
+            _unitOfWorkMock.Setup(x => x.LessonRepository).Returns(_lessonRepositoryMock.Object);
+            _unitOfWorkMock.Setup(x => x.ThemeRepository).Returns(_themeRepositoryMock.Object);
+            _unitOfWorkMock.Setup(x => x.MentorRepository).Returns(_mentorRepositoryMock.Object);
+            _unitOfWorkMock.Setup(x => x.StudentGroupRepository).Returns(_studentGroupRepositoryMock.Object);
+            _unitOfWorkMock.Setup(x => x.VisitRepository).Returns(_visitRepositoryMock.Object);
+
+            _currentUserServiceMock = GetCurrentUserAsExistingStudent();
         }
 
         [Fact]
-        public async Task CreateLesson_ReturnLessonDTo_NotEmpty()
+        public async Task CreateLesson_LessonDTO_ShouldBeNotNull()
```

Comment:
The convention of unit tests should be next MethodName_State_ShoudlExpectedResult.
Please change LessonDTO to ValidDataPassed

Reply:
done